=== modified file 'libempathy-gtk/empathy-irc-network-chooser.c'
--- old/libempathy-gtk/empathy-irc-network-chooser.c	2011-02-24 17:01:06 +0000
+++ new/libempathy-gtk/empathy-irc-network-chooser.c	2011-02-24 17:08:47 +0000
@@ -40,7 +40,7 @@
 
 #include "empathy-irc-network-chooser.h"
 
-#define DEFAULT_IRC_NETWORK "irc.gimp.org"
+#define DEFAULT_IRC_NETWORK "chat.freenode.net"
 #define DEFAULT_IRC_PORT 6667
 #define DEFAULT_IRC_SSL FALSE
 

